packages/wonder-blocks-dropdown/components/dropdown-core.js

```diff
@@ -463,8 +522,45 @@ export default class DropdownCore extends React.Component<
         }
     }
 
+    maybeRenderSearchTextInput() {
+        const {items, onSearchTextChanged, searchText} = this.props;
+
+        if (!onSearchTextChanged || typeof searchText !== "string") {
+            return null;
+        }
+
+        const noResult = onSearchTextChanged && items.length === 0;
+        // TODO(jangmi): Use translated string for "No results"
+        return (
+            <React.Fragment>
+                <SearchTextInput
+                    onChange={onSearchTextChanged}
+                    onClick={() => {
+                        this.handleClickFocus(0);
+                        this.focusCurrentItem();
+                    }}
+                    itemRef={this.state.itemRefs[0].ref}
+                    searchText={searchText}
+                    style={styles.searchInput}
+                />
+                {noResult && (
+                    <LabelMedium style={styles.noResult}>
+                        No results
+                    </LabelMedium>
+                )}
```

Comment:
What do you think about making this a `maybe...` function too rather than inlining it?